### PR TITLE
Move ceremony command to R2

### PR DESCRIPTION
**Pull request by danield9tqh**
## Summary
For cost saving we are moving future trusted setup ceremonies to use Cloudflare R2 instead of S3

## Testing Plan
Locally tested a few contributions

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/src/commands/service/ceremony.ts [hunk 1]
```diff
@@ -29,6 +29,14 @@ export default class Ceremony extends IronfishCommand {
       description: 'S3 bucket to download and upload params to',
       default: 'ironfish-contributions',
     }),
+    downloadPrefix: Flags.string({
+      char: 'b',
+      parse: (input: string) => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Prefix for contribution download URLs',
+      // TODO: update this to non-dev endpoint to avoid rate limiting
+      default: 'https://pub-6a239e04e140459087cf392ffc3245b1.r2.dev',
```
Comment (dguenther) [inline ironfish-cli/src/commands/service/ceremony.ts:38]: This isn't the bucket, right? I assume it's like an account name

**Review by dguenther — APPROVED** [review]

File: ironfish-cli/src/utils/s3.ts [hunk 2]
```diff
@@ -309,6 +318,31 @@ export function getS3Client(
...
+}): S3Client {
+  return new S3Client({
+    region: 'auto',
+    endpoint: R2_ENDPOINT,
+    credentials: {
+      accessKeyId: credentials.r2AccessKeyId,
+      secretAccessKey: credentials.r2SecretAccessKey,
+    },
+  })
+}
```
Comment (NullSoldier) [inline ironfish-cli/src/utils/s3.ts:333]: These are binary compatible with S3, so I wouldnt expect this to exist. I would just expect these to be parameters that you pass into getS3Client but we can refactor this later.

**Review by NullSoldier — APPROVED** [review]

File: ironfish-cli/src/commands/service/ceremony.ts (hunk 1 above)
Comment (ygao76) [reply ironfish-cli/src/commands/service/ceremony.ts:38]: its the dev testing url. We will replace it with ironfish.network domain name when close to launch.

File: ironfish-cli/src/utils/s3.ts [hunk 3]
```diff
@@ -309,6 +318,31 @@ export function getS3Client(
...
+
+export async function getR2Credentials(): Promise<R2Secret | undefined> {
+  const client = new SecretsManagerClient({ region: 'us-east-1' })
```
Comment (ygao76) [inline ironfish-cli/src/utils/s3.ts:336]: I am not sure about this region config. If the ceremony server has terraform config, it prob wont need it. 
Maybe try it without region on the github workflow

File: ironfish-cli/src/commands/service/ceremony.ts (hunk 1 above)
Comment (danield9tqh) [reply ironfish-cli/src/commands/service/ceremony.ts:38]: Yea that is basically a link to the bucket

**Pull request merged**